api/v1beta1/openstackdataplanenodeset_webhook.go

```diff
@@ -189,38 +171,62 @@ func (r *OpenStackDataPlaneNodeSet) ValidateUpdate(old runtime.Object) (admissio
 			fmt.Errorf("expected a OpenStackDataPlaneNodeSet object, but got %T", oldNodeSet))
 	}
 
+	errors := r.Spec.ValidateUpdate(&oldNodeSet.Spec)
+
+	if errors != nil {
+		openstackdataplanenodesetlog.Info("validation failed", "name", r.Name)
+		return nil, apierrors.NewInvalid(
+			schema.GroupKind{Group: "dataplane.openstack.org", Kind: "OpenStackDataPlaneNodeSet"},
+			r.Name,
+			errors,
+		)
+	}
+
+	return nil, nil
+}
+
+func (r *OpenStackDataPlaneNodeSetSpec) ValidateUpdate(oldSpec *OpenStackDataPlaneNodeSetSpec) field.ErrorList {
+
 	var errors field.ErrorList
 	// Some changes to the baremetalSetTemplate after the initial deployment would necessitate
 	// a redeploy of the node. Thus we should block these changes and require the user to
 	// delete and redeploy should they wish to make such changes after the initial deploy.
 	// If the BaremetalSetTemplate is changed, we will offload the parsing of these details
 	// to the openstack-baremetal-operator webhook to avoid duplicating logic.
-	if !reflect.DeepEqual(r.Spec.BaremetalSetTemplate, oldNodeSet.Spec.BaremetalSetTemplate) {
+	if !reflect.DeepEqual(r.BaremetalSetTemplate, oldSpec.BaremetalSetTemplate) {
 
 		// Call openstack-baremetal-operator webhook Validate() to parse changes
-		err := r.Spec.BaremetalSetTemplate.Validate(oldNodeSet.Spec.BaremetalSetTemplate)
+		err := r.BaremetalSetTemplate.Validate(oldSpec.BaremetalSetTemplate)
 		if err != nil {
 			errors = append(errors, field.Forbidden(
 				field.NewPath("spec.baremetalSetTemplate"),
 				fmt.Sprintf("%s", err)))
 		}
 	}
 
-	if errors != nil {
+	return errors
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *OpenStackDataPlaneNodeSet) ValidateDelete() (admission.Warnings, error) {
+	openstackdataplanenodesetlog.Info("validate delete", "name", r.Name)
+	errors := r.Spec.ValidateDelete()
+
+	if len(errors) != 0 {
+		openstackdataplanenodesetlog.Info("validation failed", "name", r.Name)
+
 		return nil, apierrors.NewInvalid(
-			schema.GroupKind{Group: "dataplane.openstack.org", Kind: "OpenStackDataPlaneNodeSet"},
+			schema.GroupKind{Group: "dataplane.openstack.org", Kind: "OpenStackDataplaneNodeSet"},
```

Comment:
We probably need to keep capitalization consistent. As far as I see the Plane is capitalized in other places:
https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/PROJECT#L20 
https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/config/crd/bases/dataplane.openstack.org_openstackdataplanenodesets.yaml#L12

Reply:
Done.